### DISCUSSION TITLE
[Feedback Wanted] Goilerplate - Modern UI Components for Go & Templ

### DISCUSSION THREAD

**Discussion by axadrn** [a-h/templ#951] [Show and tell]
Hey fellow developers,

I'm working on Goilerplate, a library of modern UI components for Go and Templ. It's designed to help build sleek, interactive web applications, with seamless Alpine.js integration for enhanced client-side functionality.

Currently, I have only a few components implemented, with plans for daily updates and bug fixes. Before investing more time, I'd love to get your thoughts:

* Is this something you'd find useful in your Go projects?
* What components would you like to see prioritized?
* Any features you think are crucial for a project like this?

Your feedback will help shape the direction of Goilerplate. Thanks in advance!

Check it out: https://goilerplate.com/

**Comment by a-h** [comment]
Looks great!

**Reply by axadrn** [reply]
Thank you!

**Comment by axadrn** [comment]
Hey fellow developers,

Following up on Goilerplate's recent v0.20.1 release, I wanted to share some exciting new developments that make it even easier to get started with and use our UI component library for Go and Templ.

1. **New Quickstart Project**
We've created a [Goilerplate Quickstart](https://github.com/axzilla/goilerplate-quickstart) repository to help you hit the ground running. It's pre-configured with Goilerplate, Templ, and Tailwind CSS, providing a solid foundation for your projects.

2. **Flexible Usage Options**
   - **Package Library**: `go get github.com/axzilla/goilerplate`
   - **Copy-Paste Components**: For maximum customization, you can now easily copy individual components into your project.
   - **Mix and Match**: Use both approaches as needed in your project!

3. **Lucide Icon Wrapper**
We've integrated a Lucide icon wrapper, giving you access to a wide range of beautiful, customizable icons right out of the box.

Here's a quick example of how easy it is to use Goilerplate now:

```go
import "github.com/axzilla/goilerplate/pkg/components"

// In your Templ files
@components.Button(components.ButtonProps{Text: "Click me"})
@components.Icon(components.IconProps{Name: "user"})
```

We'd love to hear your thoughts on these new features. Have you tried the Quickstart project? How do you prefer to use UI libraries in your Go projects?

Check out our [GitHub repo](https://github.com/axzilla/goilerplate) for more details and don't hesitate to open issues or contribute!

Let's discuss how we can make Go UI development even better!
@components.Icon(components.IconProps{Name: "user"})
We'd love to hear your thoughts on these new features. Have you tried the Quickstart project? How do you prefer to use UI libraries in your Go projects?

Check out our [GitHub repo](https://github.com/axzilla/goilerplate) for more details and don't hesitate to open issues or contribute!

Let's discuss how we can make Go UI development even better!

https://goilerplate.com/

**Reply by a-h** [reply]
I like it. I would love to see some layouts, especially a 3 column layout (global nav, content, local nav) that drops the 1st and 3rd columns for smaller screens, and introduces a hamburger menu to bring in the global nav.

I have something like this in my latest project:

```templ
templ Page(left, middle, right templ.Component) {
	<!DOCTYPE html>
	<html>
		<head>
			<meta charset="UTF-8"/>
			<meta name="viewport" content="width=device-width, initial-scale=1.0"/>
			<title>Content</title>
			<link rel="stylesheet" href="/static/modern-normalize.css"/>
			<link rel="stylesheet" href="/static/custom.css"/>
			<link rel="stylesheet" href="/static/sakura-fragments.css"/>
			<script src="/static/htmx.min.js" integrity="sha384-Y7hw+L/jvKeWIRRkqWYfPcvVxHzVzn5REgzbawhxAuQGwX1XWe70vji+VSeHOThJ"></script>
			<script src="/static/sse.js" integrity="sha384-fw+eTlCc7suMV/1w/7fr2/PmwElUIt5i82bi+qTiLXvjRXZ2/FkiTNA/w0MhXnGI"></script>
		</head>
		<body>
			<div class="layout">
				<div class="sidebar-left">
					@left
				</div>
				<div class="content">
					<a id="top"></a>
					<div class="content-scroll-container">
						@middle
					</div>
				</div>
				<div class="sidebar-right">
					@right
				</div>
			</div>
		</body>
	</html>
}
```

Some sort of recursive menu, that allows you to set the max depth?

```templ
templ menu(items []site.MenuItem) {
	<ul>
		for _, item := range items {
			<li>
				<a href={ templ.SafeURL(item.URL) }>{ item.Title }</a>
				@menu(item.Children)
			</li>
		}
	</ul>
}
```

Interested in additional form components, and some docs on how to make your own components that match the style of goilerplate well.

Think the icons are very cool. I could see myself using them. The docs should probably link to Lucide so you can see what icons you can use. Ideally, the Icon name would be a constant so you get autocomplete. Also, it looks like the icons (internally) are stored in a map, which means that they're all stored in RAM, even if you don't use them in your project? I was wondering if they were constants, whether they would be optimised out of the final binary and not make it into RAM.

```templ
@icons.Layers2(props...)
```

**Reply by axadrn** [reply]
Hey @a-h, thanks a ton for your feedback! I've been busy working on some of the things you mentioned:

* The icon wrapper's been updated and is now in the quickstart repo.
* Form components are next on my list.
* I'm planning to add that extended layout with navbar and menu to the quickstart project soon.

Your ideas are really helping shape Goilerplate. Thanks for taking the time to share them!

**Reply by axadrn** [reply]
BTW: Can you please explain these two topics in more detail?

> Interested in additional form components, and some docs on how to make your own components that match the style of 
> goilerplate well.

**Reply by a-h** [reply]
Sorry, realised I never replied, but got pinged on this thead and saw it!

I was just saying that I would be interested in seeing more form components - checkboxes, date pickers, etc.

Re: matching style, I was looking for some sort of style guide for extending the set of goilerplate components. For example, explanations of which colours / fonts / padding / icons and css classes etc. to use if you want to create something that matches the visual style of the component set.

**Reply by axadrn** [reply]
Got it, thanks for that. Did you already check out the "new" icons component?

**Reply by a-h** [reply]
This one? https://goilerplate.com/docs/components/icon

Looks great!

**Reply by axadrn** [reply]
Yes thanks. It's a "typed" and memory saving version now.

**Comment by vinicius404** [comment]
This is really interesting. Have you looked into [https://goship.it/](https://goship.it/)?

**Reply by axadrn** [reply]
Yep, I know about this lib. Good to have options in this space!

**Comment by axadrn** [comment]
🎉 Goilerplate v0.22.0 is out!

✨ New Components:
- Textarea
- Slider

💎 Improvements:
- Better Alpine.js integration
- Enhanced form components
- Simplified UI structure
- Improved accessibility

🔍 Check it out:
https://goilerplate.com
https://github.com/axzilla/goilerplate

#golang #webdev #UILibrary

**Comment by axadrn** [comment]
Great news everyone! We just launched preview.goilerplate.com where you can now test and use all our newest features and components before they hit the main release! 🚀

It's a rolling release regularly synced from our dev branch, and you can start using it right away - either by copy/pasting components or by installing via `go get github.com/axzilla/goilerplate@preview`

Check it out and let me know what you think! 😊


https://github.com/user-attachments/assets/c2c5d7f7-e8f7-472e-843c-8fc6502d9789

**Comment by axadrn** [comment]
🔥 **templUI v0.23.0** is here!  
✨ No-Node.js setup  
📚 Revamped docs & quickstart  
🧩 More accessible form components & modular structure  
📅 Enhanced datepicker with custom placeholders & formats  
🎨 Theme previews & updated icons

Check it out 👉 [templui.io](https://templui.io)

**Comment by andrewshadura** [comment]
It’d probably make sense to prominently mention this in the original post:
**[Goilerplate is now templUI - Update on UI Components for Templ](https://github.com/a-h/templ/discussions/1014)**

**Reply by axadrn** [reply]
Hey, do you mean I should delete the Goilerplate -> templUI post?

**Reply by andrewshadura** [reply]
No, just edit this one and add the link 🙂

**Comment by axadrn** [comment]
# Goilerplate is now templUI - Update on UI Components for Templ

Hi Templ community! 👋

Quick update: Goilerplate has evolved and is now templUI! 

## Why the new name?

The rebranding better reflects what we're building - a professional UI component library specifically designed for Go and Templ projects. While the name has changed, our mission remains the same: providing type-safe, beautiful components for the Templ ecosystem.

## What's new since last time?

- Fresh branding and new website: templui.io
- Improved documentation
- More components and examples
- Better Alpine.js integration
- Enhanced Tailwind customization

## Next Steps

We're continuing to build and improve templUI. The project is still a work in progress, but we're excited about the direction and the feedback we've received from the community.

## Links

- [GitHub: github.com/axzilla/templui](github.com/axzilla/templui)
- [Website: templui.io](templui.io)
- [Discussion](https://github.com/a-h/templ/discussions/1014)

Thank you all for your continued support and feedback! Looking forward to building more together. 🚀
